Streetcode/Streetcode.BLL/MediatR/Newss/GetNewsAndLinksByUrl/GetNewsAndLinksByUrlHandler.cs

```diff
@@ -0,0 +1,108 @@
+﻿using AutoMapper;
+using FluentResults;
+using MediatR;
+using Streetcode.BLL.DTO.News;
+using Streetcode.BLL.Interfaces.BlobStorage;
+using Streetcode.DAL.Entities.News;
+using Streetcode.DAL.Repositories.Interfaces.Base;
+using Microsoft.EntityFrameworkCore;
+
+namespace Streetcode.BLL.MediatR.Newss.GetNewsAndLinksByUrl
+{
+    public class GetNewsAndLinksByUrlHandler : IRequestHandler<GetNewsAndLinksByUrlQuery, Result<NewsDTOWithURLs>>
+    {
+        private readonly IMapper _mapper;
+        private readonly IRepositoryWrapper _repositoryWrapper;
+        private readonly IBlobService _blobService;
+        public GetNewsAndLinksByUrlHandler(IMapper mapper, IRepositoryWrapper repositoryWrapper, IBlobService blobService)
+        {
+            _mapper = mapper;
+            _repositoryWrapper = repositoryWrapper;
+            _blobService = blobService;
+        }
+
+        public async Task<Result<NewsDTOWithURLs>> Handle(GetNewsAndLinksByUrlQuery request, CancellationToken cancellationToken)
+        {
+            string url = request.url;
+            var newsDTO = _mapper.Map<NewsDTO>(await _repositoryWrapper.NewsRepository.GetFirstOrDefaultAsync(
+                predicate: sc => sc.URL == url,
+                include: scl => scl
+                    .Include(sc => sc.Image)));
+
+            if (newsDTO is null)
+            {
+                return Result.Fail($"No news by entered Url - {url}");
+            }
+
+            if (newsDTO.Image is not null)
+            {
+                newsDTO.Image.Base64 = _blobService.FindFileInStorageAsBase64(newsDTO.Image.BlobName);
+            }
+
+            var news = (await _repositoryWrapper.NewsRepository.GetAllAsync()).ToList();
```

Comment:
I think it is better to send here the predicate with the check of the id range because all data is loaded to memory now.

Reply:
I don't see any changes that can be done here bcs there is a variety of situations that could happen, the first problem is that I can't possibly know what news should I take bcs I can't have an index of my current news without having the list of all news, so I can't check if this is the first news in the DB or the last so there will be problems with checking these things.

Reply:
the first news in the list could have id 1 and the second will have id 10 so I can't know in advance which specific range of values I should take